### PR TITLE
Filtering all the things

### PR DESCRIPTION
**Pull request by cjjaeger**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -18,10 +18,10 @@ class App extends Component {
                 "zip": "",
                 "inArea": false,
                 "radius": null,
-                "tuition":[0,70000],
-                "SAT":null,
-                "ACT":null,
-                "ranking": null,
+                "tuition":[0,53000],
```
Comment (zmbc) [inline src/App.js]: Why the change? Are there schools more expensive than 53000?

File: src/App.js [hunk 2]
```diff
@@ -18,10 +18,10 @@ class App extends Component {
...
+                "SAT":"",
+                "ACT":"",
+                "ranking": "",
```
Comment (zmbc) [inline src/App.js]: Aren't all three of SAT, ACT, and Ranking numbers?

Also, why not change all these property names to use no quotes, e.g.
```
ACT: "",
SAT: "",
...
```

File: src/App.js [hunk 3]
```diff
@@ -34,50 +34,130 @@ class App extends Component {
...
 
         this.handleChange(event);
     }
     changeValue(e){
         this.state.filter.tuition = e.target.value;
-        console.log(this.state);
     }
     applyFilter(e){
         e.preventDefault();
+        if(/(^\d{5}$)|(^\d{5}-\d{4}$)/.test(this.state.filter.zip)){
```
Comment (zmbc) [inline src/App.js]: Put spaces before the `(` and before the `{`

File: src/App.js [hunk 4]
```diff
@@ -34,50 +34,130 @@ class App extends Component {
...
+            var url = "https://maps.googleapis.com/maps/api/geocode/json?address="+ encodeURIComponent(this.state.filter.zip);
+            fetch(url) //download the data
+            .then(function(res) { return res.json(); })
```
Comment (zmbc) [inline src/App.js]: Indent these `.then`s one level

File: src/App.js [hunk 5]
```diff
@@ -34,50 +34,130 @@ class App extends Component {
...
+            .then((datas) =>{
+                return this.setZipLocation(datas.results);
+            })
+            .then(this.satFilter)
+            .then(this.actFilter)
+            .then(this.rankingFilter)
+            .then(this.tuitionFilter);
+        }else{
```
Comment (zmbc) [inline src/App.js]: Put spaces around `else`

File: src/App.js [hunk 6]
```diff
@@ -34,50 +34,130 @@ class App extends Component {
...
+            this.tuitionFilter(this.rankingFilter(this.actFilter(this.satFilter(data))));
```
Comment (zmbc) [inline src/App.js]: I think there's a better way to do this if/else. Here's how I'd do it:
- Make a variable for the ZIP promise.
- Then do an if/else. If there is a ZIP code, set the ZIP promise. If not, set the ZIP promise to `Promise.resolve(data)`.
- Then outside the if/else, you should be able to use your variable with the string of `.then`s you have.

This way you won't have to repeat the set of operations you want to do twice.

File: src/App.js [hunk 7]
```diff
@@ -34,50 +34,130 @@ class App extends Component {
...
+            return datas.filter((obj)=>{
+                return this.state.filter.ACT>=obj["2014.admissions.act_scores.midpoint.cumulative"];
+            });
+        }else{
+            return datas;
+        }
+        
+    }
+    rankingFilter(datas){
+
```
Comment (zmbc) [inline src/App.js]: Don't put blank lines at the beginning and end of functions. Do put blank lines before functions.

File: src/App.js [hunk 8]
```diff
@@ -34,50 +34,130 @@ class App extends Component {
...
+            });
+            return newData;
+        }else{
+            return datas;
+        }
+
+    }
+        
+
+    tuitionFilter(stuff){
```
Comment (zmbc) [inline src/App.js]: There's probably a better name for this variable

File: src/App.js [hunk 9]
```diff
@@ -34,50 +34,130 @@ class App extends Component {
...
+                return obj['2014.cost.tuition.out_of_state'] >= this.state.filter.tuition[0] && obj['2014.cost.tuition.out_of_state'] <= this.state.filter.tuition[1];
+            });
+        var childState = Object.assign({}, this.state);
+        childState.data = newData;
+        this.child = React.cloneElement(this.props.children, childState);
+            this.setState({"data": newData});
+
+    }
+
+    distFrom( lat1,  lng1,  lat2,  lng2) {
```
Comment (zmbc) [inline src/App.js]: Weird spaces around these parameters (I realize you copy-pasted this)

File: src/DonutScatter.js [hunk 10]
```diff
@@ -88,7 +88,7 @@ var DonutScatter = function() {
 
             // Calculate x and y scales
             let xMax = d3.max(data, (d) => +d[xAccessor]) * 1.01;
-            let xMin = d3.min(data, (d) => +d[xAccessor]) * .5;
+            let xMin = d3.min(data, (d) => +d[xAccessor]) -1000;
```
Comment (zmbc) [inline src/DonutScatter.js]: This change shouldn't be here. Will probably work itself out when you reverse-merge from master.

File: src/App.js (hunk 1 above)
Comment (cjjaeger) [inline src/App.js]: no
